scripts/validate_data.py

```diff
@@ -47,14 +51,33 @@ def validate_data(data_directory):
         If hash value for any file is different from hash value recorded in
         ``data_hashes.txt`` file.
     """
-    # Read lines from ``data_hashes.txt`` file.
-    # Split into SHA1 hash and filename
-    # Calculate actual hash for given filename.
-    # If hash for filename is not the same as the one in the file, raise
-    # ValueError
-    # This is a placeholder, replace it to write your solution.
-    raise NotImplementedError('This is just a template -- you are expected to code this.')
+    # debug print statement
+    print(f"Looking for 'hash_list.txt' in directory: {data_directory}")
+
+    hash_list_path = os.path.join(data_directory, 'group-00', 'hash_list.txt')
+
+    # another debug print statement
+    print(f"Full path to 'hash_list.txt' found: {hash_list_path}")
+
+    # open file and read each line
+    try:
+        with open(hash_list_path, 'rt') as f:
+            lines = f.readlines()
+    # raising error if the file was not found
+    except FileNotFoundError:
+        print(f"Error: File 'hash_list.txt' not found in directory: {data_directory}/group-00")
+        return
+
+    #rading each line of the file
+    for line in lines:
+        expected_hash, filename = line.strip().split()
```

Comment:
Stylistic, but if you call split without separator, the strip is not necessary.
```suggestion
        expected_hash, filename = line.split()
```